qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/project/MockQuteProjectRegistry.java

```diff
@@ -101,28 +105,40 @@ public CompletableFuture<Location> getJavaDefinition(QuteJavaDefinitionParams pa
 				if (methodName != null) {
 					String getterMethodName = computeGetterName(methodName);
 					String booleanGetterName = computeBooleanGetterName(methodName);
-					JavaMethodInfo methodInfo = findMethod(classInfo, methodName, getterMethodName, booleanGetterName);
+					JavaMethodInfo methodInfo = super.findMethod(classInfo, methodName, getterMethodName,
+							booleanGetterName);
 					if (methodInfo != null) {
 						definitionRange = JAVA_METHOD_RANGE;
-					}
+					} 
 				} else {
 					// Definition for class
 					definitionRange = JAVA_CLASS_RANGE;
 				}
 			}
-
-			if (definitionRange != null) {
-				int index = className.indexOf('<');
-				if (index != -1) {
-					// ex : java.util.List<E>
-					// Remove generic
-					className = className.substring(0, index);
-				}
-				// ex : java/util/List.java
-				String javeFileUri = className.replaceAll("[.]", "/") + ".java";
-				Location location = new Location(javeFileUri, definitionRange);
-				return CompletableFuture.completedFuture(location);
+		} 
+
+		if (definitionRange == null) {
+			// Find by field, method of value resolver
+			String methodName = params.getSourceMethod();
+			if (methodName != null) {
+				JavaMethodInfo methodInfo = project.getMethodValueResolver(className, methodName);
+				if (methodInfo != null) {
+					definitionRange = JAVA_STATIC_METHOD_RANGE;
+				} 
+			}
+		}
+		
+		if (definitionRange != null) {
+			int index = className.indexOf('<');
+			if (index != -1) {
+				// ex : java.util.List<E>
+				// Remove generic
+				className = className.substring(0, index);
 			}
+			// ex : java/util/List.java
+			String javeFileUri = className.replaceAll("[.]", "/") + ".java";

```

Comment:
Could you use .replace instead in this case? See https://medium.com/javarevisited/micro-optimizations-in-java-string-replaceall-c6d0edf2ef6

Reply:
I can omptimize that, but to be honnest with you, it is used just for test. Do you think it is interesting to do that?

Reply:
just good practice to follow

Reply:
You are right, fixed. Thanks for this helpful article!